application/models/services/PermissionManager.php

```diff
@@ -185,4 +183,23 @@ public function getNewPermission()
     {
         return new Permission();
     }
+
+    /**
+     * get the current permission for the current user
+     * @param string $modelName the model name
+     * @param integer $modelId the model id
+     * @param string $sPermission the specific permission name
+     * @param string $crud the crud ('create', 'read', 'update', 'delete', 'import', 'export')
+     * @return boolean
+     */
+    public function getCurrentPermission($modelName, $modelId, $sPermission, $crud, $userId)
+    {
+        /* string : function name to use for final Permission */
+        $permissionFunctionName = "has{$modelName}Permission";
+        if(method_exists($this->permission,$permissionFunctionName)) {
+            return $this->permission->$permissionFunctionName($modelId, $sPermission, $crud, $userId);
```

Comment:
Interfaces are better than variable variables.......

Reply:
This is impossible for a static analyser to understand.

Reply:
I don't have idea on the way to do here. 

I **need** to check hasSurveysGroupsPermission or hasSurveysInGroupPermission (and hasSurveyPermission) …

Reply:
Use `instanceof`?

Reply:
Or add a new interface for models who have permission. `HasPermissionCheckInterface`.

Reply:
If the implementation is the same for all classes implementing the interface, you can use a trait. Otherwise, you will have to add the method inside each class manually.

Reply:
> Use `instanceof`?

See https://github.com/LimeSurvey/LimeSurvey/pull/1709#issuecomment-754775785 this need to move checkPermission function in model … create a really simple checkPermission in LSActive record , and replace in Survey, SurveysGroups,SurveysInGroup

> If the implementation is the same for all classes implementing the interface, you can use a trait. Otherwise, you will have to add the method inside each class manually.

- Surveys must update $crud, check Global, check SurveysInGroup, check Own
- SurveysGroups must check global + own
- SurveysInGroup  : update crud + check Global, check Own
- Template : really specific : global + $sTemplateName as object name
- Roles : just need to check global

Then : best seems to send model as LSActiverecord and call $model->getPermission new function …

Reply:
> create a really simple checkPermission in LSActive record

Yes, but please don't do this without adding an interface, to distinguish models with permission check from those that haven't.

Reply:
Arg … without interface : https://github.com/LimeSurvey/LimeSurvey/pull/1709/files#diff-ebf85ba8a486bce5d44fcf65c87a272498d2a841b68fb535e69db3eaf879ade1R485

Reply:
I don't understand : all object are already child of LSActiveRecord … you want a LSActivaveRecord without intefcae an one implement a new interface ?

Reply:
It's about _intent_. Not all classes that inherit from LSActiveRecord have permissions attached to them. Also that a class should have a clear purpose, not "contain everything" as LSActiveRecord does.

Reply:
> Arg … without interface : https://github.com/LimeSurvey/LimeSurvey/pull/1709/files#diff-ebf85ba8a486bce5d44fcf65c87a272498d2a841b68fb535e69db3eaf879ade1R485

Cool, then you can use interface + trait instead. :)

Reply:
You'd have a new interface "HasPermissionInterface" and a new trait that implements it, "HasPermissionTrait".

Reply:
Thank you